Assets/MixedRealityToolkit/Utilities/Editor/PackageManifest/PackageManifestUpdater.cs

```diff
@@ -42,10 +46,96 @@ private static string GetPackageManifestFilePath()
+            string[] versionComponents = packageVersion.Split(new char[] { '-' }, 2);
+
+            // Parse the version component.
+            string versionString = versionComponents[0].Trim(trimChars);
+            if (Version.TryParse(versionString, out version))
+            {
+                if (versionComponents.Length == 2)
+                {
+                    // Parse the float component
+                    string prereleaseString = versionComponents[1].Trim(trimChars);
+                    if (float.TryParse(prereleaseString, out prerelease))
+                    {
+                        return true;
+                    }
+                }
+                else
+                {
+                    prerelease = 0f;
+                    return true;
+                }
+            }
+
+            version = null;
+            prerelease = float.NaN;
+            return false;
+        }
+
         /// <summary>
         /// Reports whether or not the appropriate version of MSBuild for Unity is specified
         /// in the Unity Package Manager manifest.
         /// </summary>
+        /// <param name="minPackageVersion">The minimum version of the package, as listed in the manifest.</param>
+        /// <param name="packageVersion">The version of the package, as listed in the manifest.</param>
+        /// <returns>
+        /// True if an appropriate verson of MS Build for Unity is configured in the manifest, otherwise false.
+        /// </returns>
+        internal static bool IsAppropriateMBuildVersion(string minPackageVersion, string packageVersion)
+        {
+            // Get the version of the package.
+            // Note: The version is in the following format Major.Minor.Revision[-Date.Build]
+
+            Version minVersion;
+            float minPrerelease;
+
+            // Get the min version
+            if (!TryGetVersionComponents(minPackageVersion, out minVersion, out minPrerelease))
+            {
+                return false;
+            }
+
+            // Get the current version from the manifest
+            Version currentVersion;
+            float currentPrerelease;
+            if (!TryGetVersionComponents(packageVersion, out currentVersion, out currentPrerelease))
+            {
+                return false;
+            }
+
+            // Evaluate the results.
+            // * (currentVersion > minVersion)                                                                  return true;
+            // * (currentVersion == minVersion && currentPrerelease == minPrerelease)                           return true;
+            // * (currentVersion == minVersion && minPrerelease != 0 && currentPrerelease >= minPrerelease)   return true;
+            // * all other combinatons                                                                          return false;
+            if (currentVersion > minVersion)
+            { return true; }
```

Comment:
if ()
{
    return true;
}

Here and below

Reply:
will do.

Reply:
done